src/commands/ban.ts

```diff
@@ -0,0 +1,128 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptionsRunTypeEnum } from '@sapphire/framework';
+import {
+  EmbedBuilder,
+  PermissionFlagsBits,
+  type ChatInputCommandInteraction,
+} from 'discord.js';
+import { Color } from '../lib/embeds.js';
+import { messageLogger, moderationLogs } from '../index.js';
+import type { banLog } from '../lib/moderation.js';
+
+@ApplyOptions<Command.Options>({
+  description: 'Ban user',
+  requiredClientPermissions: [PermissionFlagsBits.BanMembers],
+  requiredUserPermissions: [PermissionFlagsBits.BanMembers],
+
+  runIn: [CommandOptionsRunTypeEnum.GuildAny],
+})
+export class BanCommand extends Command {
+
+  public override async chatInputRun(interaction: ChatInputCommandInteraction) {
+    if (!interaction.inGuild()) {
+      return;
+    }
+    const user = interaction.options.getUser(Option.User, true);
+    const reason = interaction.options.getString(Option.Reason);
```

Comment:
If we want reason to be required, we can make it required on the frontend, so that the command can't even be submitted without a reason.
```suggestion
    const reason = interaction.options.getString(Option.Reason, true);
```